### ISSUE TITLE
Fix missing listitem role in the Connectors list

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#82659]
### Description

Discovered while tedsting Core https://github.com/WordPress/wordpress-develop/pull/13348

This is a blocker for Core-66015 and should be addressed soon, please.

Can we please make sure this fix will be part of the changes for WordPress 7.1.1?
cc @adamsilverstein @adrianduffell @andreiglingeanu

In the Connectors page, the list of connectors uses [a VStack component with a hardcoded role='list'](https://github.com/WordPress/gutenberg/blob/e555de6aea0cf10b2f4381fa4f4f221626418aad/routes/connectors-home/stage.tsx#L148).

When using a `role='list'` all children *must* have a `role='listitem'`.

The missing listitem role is caught by the Axe-core automated checks we want to introduce in Core and reported as a violation.

Either the hardcoded `role='list' should be removed or [the ConnectorItem component](https://github.com/WordPress/gutenberg/blob/0eaf6c57c664895b52508eb1c96ff2d6336cbb63/routes/connectors-home/default-connectors.tsx#L226) should have a `role='listitem'`.

For the future, it is important to note that any piece of UI added into the list of connectors *must* be a listitem.

Note:
I can't see a GitHub label for `packages/connectors` or `routes`.

### Step-by-step reproduction instructions

N/A

### Screenshots, screen recording, code snippet

<img width="1469" height="972" alt="Image" src="https://github.com/user-attachments/assets/4b174d47-9362-4fcc-9f04-e3919263e992" />

_No response_

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure
